### ISSUE TITLE
New Source: Zendesk Talk

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#2346]
## Tell us about the new integration you’d like to have
A user mentioned offline that they'd like to pull data from the [Zendesk Talk API](https://developer.zendesk.com/rest_api/docs/voice-api/introduction) via Airbyte.

## Describe the context around this new integration
TODO

## Describe the alternative you are considering or using
Building a custom pipeline

**Comment by sherifnada** [comment]
@kitkatbarxp some questions to help us get started: 

* which endpoints from the Zendesk Talk API do you need? 
* which destinations are you sending this data to ?
* can you describe the usecase/context around using this API?

**Comment by kitkatbarxp** [comment]
endpoints: incremental exports

destination: it'd likely be a postgres instance for now. In the medium/long term, there are possibilities of loading the data into redshift/snowflake, or at least some type of json file in s3.

context/use case: My company uses Zendesk heavily to support our customers. We support them via chat, email, and phone. There are a lot of operational and customer insights that we can derive to help us support our customers better and also improve our own product. Specifically from the incremental exports API, we can look at hold time, call time, associate the calls to specific Zendesk tickets to paint a more complete picture of support.

**Comment by kitkatbarxp** [comment]
On the incremental exports endpoint, both Incremental Calls and Incremental Call Legs would be super useful.

**Comment by danieldiamond** [comment]
Hi @sherifnada, 
Was glad to stumble upon this issue. Considering implementing airbyte and including the Talk API in the zendesk connector would be a massive pull to get me started.

Destination: Redshift initially but looking to migrate to snowflake in the short future.

Context/Use-Case: Similarly to @kitkatbarxp use-cases and aside from all the usual meaningful insights, the most pressing requirement for call data is to provide a data driven approach to staff scheduling (understanding inbound call volumes, peak times) and improving customer care by reducing calls to voicemail. etc.

Thanks in advance!

**Comment by sherifnada** [comment]
@danieldiamond that's a really interesting use case, thanks for sharing the context! Will see what we can do on our side to prioritize, will start investigating soon. 

So as far as I can tell, the most important endpoints for you @danieldiamond and @kitkatbarxp would be: 
* [incremental export calls](https://developer.zendesk.com/rest_api/docs/voice-api/incremental_exports#incremental-calls-export)
* [incremental export call legs](https://developer.zendesk.com/rest_api/docs/voice-api/incremental_exports#incremental-call-legs-export)
* [account overview stats](https://developer.zendesk.com/rest_api/docs/voice-api/stats#show-account-overview)

LMK if any other endpoints are useful

**Comment by vitaliizazmic** [comment]
## Integration Vetting
The integration has already been vetted in https://docs.google.com/spreadsheets/d/1CUkw5ha0zPGJT1-vhe2rT3oIWt_6rVieH40J9fVrLJM/edit#gid=0 It's ready for implementation.

**Webhook-based? (no/partially/yes)**
No

**Available authentication modes (API key/Oauth/other)**
login/password, API Token

**Creating an account**
Sandbox

**How to populate the account with data?**
Account can be populated with data? Yes

**Available streams for sync**

- phone_numbers
- addresses
- greeting_categories
- greetings
- callback_requests
- account_overview
- agents_activity
- agents_overview
- current_queue_activity
- calls
- call_legs
- ivr
- ivr_menu
- ivr_routes

 
**Integration supports incremental sync?**
Yes

**Comment by yevhenii-ldv** [comment]
## Task breakdown
- [ ] Populate source with data (estimation 5-6 h).
- [ ] Create source with all streams and full table sync method (estimation 8h)
- [ ] Update source with incremental sync method (estimation 7h)
- [ ] Ensure that connector follows the best practices, rate limiting, retry call, proper error messages, has tests (estimation 4h).

**Comment by danieldiamond** [comment]
🤦  apologies for the above spam, forking repo locally and rebasing to fix open branch `ykurochkin/connector-zendesk-talk`

**Comment by yevhenii-ldv** [comment]
Hello, @danieldiamond 
Thank you for your attentiveness and responsiveness. We did have some minor problems with the project build, but they were resolved, I merged the changes from the master to the Zendesk Talk branch and the build was successful.
I published the connector image to docker hub and merged the code into the master branch. 
I suggest you try how it works and leave a comment about your experience :smiley:

Enjoy :wink:
